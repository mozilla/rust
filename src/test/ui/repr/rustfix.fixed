// run-rustfix
#[repr(align = r#"1"#)] //~ ERROR incorrect `repr(align)` attribute format
                        //~| ERROR incorrect `repr(align)` attribute format
struct A;

#[repr(align = r###"foo"###)] //~ ERROR incorrect `repr(align)` attribute format
                              //~| ERROR incorrect `repr(align)` attribute format
struct B;

#[repr(align = 1)] //~ ERROR incorrect `repr(align)` attribute format
                   //~| ERROR incorrect `repr(align)` attribute format
struct C;

#[repr(C)] //~ ERROR invalid `repr(C)` attribute: no arguments expected
                        //~| ERROR invalid `repr(C)` attribute: no arguments expected
struct D;

#[repr(C)] //~ ERROR invalid `repr(C)` attribute: no arguments expected
             //~| ERROR invalid `repr(C)` attribute: no arguments expected
struct E;

#[repr(align("1"))] //~ ERROR invalid `repr(align)` attribute
                    //~| ERROR invalid `repr(align)` attribute
struct F;

#[repr(align(r############"1"############))] //~ ERROR invalid `repr(align)` attribute
                                             //~| ERROR invalid `repr(align)` attribute
struct G;

fn main() {}
