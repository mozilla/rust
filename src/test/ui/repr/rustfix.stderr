error[E0693]: incorrect `repr(align)` attribute format
  --> $DIR/rustfix.rs:2:8
   |
LL | #[repr(align = r#"1"#)]
   |        ^^^^^^^^^^^^^^
   |
help: use parentheses instead
   |
LL | #[repr(align(1))]
   |             ^ ^

error[E0693]: incorrect `repr(align)` attribute format
  --> $DIR/rustfix.rs:6:8
   |
LL | #[repr(align = r###"foo"###)]
   |        ^^^^^^^^------------
   |                |
   |                must be a non-negative number
   |
help: use parentheses instead
   |
LL | #[repr(align(foo))]
   |             ^   ^

error[E0693]: incorrect `repr(align)` attribute format
  --> $DIR/rustfix.rs:10:8
   |
LL | #[repr(align = 1)]
   |        ^^^^^^^^^
   |
help: use parentheses instead
   |
LL | #[repr(align(1))]
   |             ^ ^

error[E0589]: invalid `repr(C)` attribute: no arguments expected
  --> $DIR/rustfix.rs:14:8
   |
LL | #[repr(C(1, "", true))]
   |        ^^^^^^^^^^^^^^
   |
help: remove additional values
   |
LL | #[repr(C)]
   |        --

error[E0589]: invalid `repr(C)` attribute: no arguments expected
  --> $DIR/rustfix.rs:18:8
   |
LL | #[repr(C())]
   |        ^^^
   |
help: remove additional values
   |
LL | #[repr(C)]
   |        --

error[E0589]: invalid `repr(align)` attribute
  --> $DIR/rustfix.rs:22:14
   |
LL | #[repr(align("1"))]
   |              ^^^
   |
help: remove the quotes
   |
LL | #[repr(align(1))]
   |             ---

error[E0589]: invalid `repr(align)` attribute
  --> $DIR/rustfix.rs:26:14
   |
LL | #[repr(align(r############"1"############))]
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: remove the quotes
   |
LL | #[repr(align(1))]
   |             ---

error[E0693]: incorrect `repr(align)` attribute format
  --> $DIR/rustfix.rs:2:8
   |
LL | #[repr(align = r#"1"#)]
   |        ^^^^^^^^^^^^^^
   |
help: use parentheses instead
   |
LL | #[repr(align(1))]
   |             ^ ^

error[E0693]: incorrect `repr(align)` attribute format
  --> $DIR/rustfix.rs:6:8
   |
LL | #[repr(align = r###"foo"###)]
   |        ^^^^^^^^------------
   |                |
   |                must be a non-negative number
   |
help: use parentheses instead
   |
LL | #[repr(align(foo))]
   |             ^   ^

error[E0693]: incorrect `repr(align)` attribute format
  --> $DIR/rustfix.rs:10:8
   |
LL | #[repr(align = 1)]
   |        ^^^^^^^^^
   |
help: use parentheses instead
   |
LL | #[repr(align(1))]
   |             ^ ^

error[E0589]: invalid `repr(C)` attribute: no arguments expected
  --> $DIR/rustfix.rs:14:8
   |
LL | #[repr(C(1, "", true))]
   |        ^^^^^^^^^^^^^^
   |
help: remove additional values
   |
LL | #[repr(C)]
   |        --

error[E0589]: invalid `repr(C)` attribute: no arguments expected
  --> $DIR/rustfix.rs:18:8
   |
LL | #[repr(C())]
   |        ^^^
   |
help: remove additional values
   |
LL | #[repr(C)]
   |        --

error[E0589]: invalid `repr(align)` attribute
  --> $DIR/rustfix.rs:22:14
   |
LL | #[repr(align("1"))]
   |              ^^^
   |
help: remove the quotes
   |
LL | #[repr(align(1))]
   |             ---

error[E0589]: invalid `repr(align)` attribute
  --> $DIR/rustfix.rs:26:14
   |
LL | #[repr(align(r############"1"############))]
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: remove the quotes
   |
LL | #[repr(align(1))]
   |             ---

error: aborting due to 14 previous errors

Some errors have detailed explanations: E0589, E0693.
For more information about an error, try `rustc --explain E0589`.
